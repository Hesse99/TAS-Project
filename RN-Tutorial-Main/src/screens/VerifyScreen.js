import React, {useState, useEffect} from 'react';
import{View, Text, Button} from 'react-native';

const VerifyScreen = () => {
    const [status, setStatus] = useState('');

    /*useEffect(() => {
      setTimeout(() => {
        setStatus('timeout is called');
      }, 5000);
    }, []);*/

    return (
      <View>
        <Text testID="myText">{status}</Text>
        <Button
          testID="myButton" 
          onPress={() => setStatus('button pressed')} 
          title="PressMe" 
        />  
      </View>
    );
};

export default VerifyScreen;