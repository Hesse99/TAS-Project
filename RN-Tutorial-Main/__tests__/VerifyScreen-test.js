import React from 'react';
import {create, act} from 'react-test-renderer';
import VerifyScreen from '../src/screens/VerifyScreen';

const tree = create(<VerifyScreen />)

jest.runAllTimers();

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('button press', () => {
  const button = tree.root.findByProps({testID: 'myButton'}).props;
  act(() => button.onPress());
  const text = tree.root.findByProps({testID: 'myText'}).props;
  expect(text.children).toEqual('button pressed');
});
