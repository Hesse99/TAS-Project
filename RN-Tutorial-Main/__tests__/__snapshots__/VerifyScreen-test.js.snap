// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View>
  <Text
    testID="myText"
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        PressMe
      </Text>
    </View>
  </View>
</View>
`;
